src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeEventService.kt

```diff
@@ -20,6 +21,7 @@ class BarcodeEventService(
   private val cjsmService: CjsmService,
 ) {
 
+  @Transactional
```

Comment:
As part of writing the tests for ContactResource / ContactRepository I found that I'd not defined the transaction boundaries correctly (at all!) in ContactService
So I've been through the other services and add the tx boundaries to the relevant methods as well 👍

Reply:
Can we have a quick chat about this - not sure we should add transactional where it's not actually needed? For example we need it for the CJSM upload hence there is a test fails without @Transactional - no similar tests for the other use cases. (Also I think you can just add @Transactional to the class and Spring works out which are readonly - [readonly is jsut a hint](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html#readOnly--) )

Reply:
OK, I'll revert my tx related changes. Reading around the subject has made me question my own understanding of when to use `@Transactional` , so I need to do some more reading up / schooling in this respect 😁